addons/stock/models/stock_quant.py

```diff
@@ -406,8 +406,7 @@ def _update_available_quantity(self, product_id, location_id, quantity, lot_id=N
         if lot_id and quantity > 0:
             quants = quants.filtered(lambda q: q.lot_id)
 
-        incoming_dates = [d for d in quants.mapped('in_date') if d]
-        incoming_dates = [fields.Datetime.from_string(incoming_date) for incoming_date in incoming_dates]
+        incoming_dates = [fields.Datetime.from_string(quant.in_date) for quant in quants if quant.in_date and quant.quantity > 0]
```

Comment:
you don't need `fields.Datetime.from_string()`, `in_date` is already a `datetime` object. And use `fields.Float.compare`

`incoming_dates = [quant.in_date for quant in quants if quant.in_date and fields.Float.compar(quant.quantity, 0, precision_rounding=quant.product_uom_id.rounding) > 0]`